src/cursor/cur_std.c

```diff
@@ -1290,6 +1290,46 @@ __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
     API_END_RET_STAT(session, ret, cursor_bound);
 }
 
+/*
+ * __wt_cursor_bounds_save --
+ *     Save cursor bounds to restore the state.
+ */
+void
+__wt_cursor_bounds_save(WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *state)
+{
+    WT_ITEM_SET(state->lower_bound, cursor->lower_bound);
+    WT_ITEM_SET(state->upper_bound, cursor->upper_bound);
+    /* Save the bound flags to the state. */
+    state->bound_flags = F_MASK(cursor, WT_CURSTD_BOUND_ALL);
+}
+
+/*
+ * __wt_cursor_bounds_restore --
+ *     Restore the cursor's bounds state. We want to change only related flags as we can't guarantee
+ *     the initial flag state of the primary and secondary column groups are the same.
+ */
+int
+__wt_cursor_bounds_restore(
+  WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_CURSOR_BOUNDS_STATE *bounds_state)
+{
+    WT_DECL_RET;
+
+    /* Clear all the bound flags. */
+    F_CLR(cursor, WT_CURSTD_BOUND_ALL);
+    /* Set the saved bound flags back to the cursor. */
+    F_SET(cursor, bounds_state->bound_flags);
+
+    if ((ret = __wt_buf_set(session, &cursor->lower_bound, bounds_state->lower_bound.data,
+           bounds_state->lower_bound.size)) != 0)
+        WT_RET_PANIC(session, ret, "Failed to restore the bounds.");
```

Comment:
Please update to: "Unrecoverable error encountered while restoring bounds"

Reply:
[98bef0d](https://github.com/wiredtiger/wiredtiger/pull/8265/commits/98bef0df873c5da8d381fe252b1eeb2f2c3a5273)